### ISSUE TITLE
`no-restricted-properties`: does not work with destructuring

### ISSUE THREAD

**Issue by ljharb** [eslint/eslint#7147]
**Tell us about your environment**

* **ESLint Version:**
v3.5.0
* **Node Version:**
v6.5.0
* **npm Version:**
3.something

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
n/a

**What did you do? Please include the actual source code causing the issue.**
```sh
echo 'const { bar } = foo; const bar = foo.bar;' | ./node_modules/.bin/eslint --stdin --parser-options '{"ecmaVersion": 6}' --rule 'no-restricted-properties:[2,{"object":"foo","property":"bar"}]' --rule no-undef:0 --rule no-unused-vars:0 --no-eslintrc
```

**What did you expect to happen?**
```sh
<text>
  1:##  error  'foo.bar' is restricted from being used  no-restricted-properties
  1:34  error  'foo.bar' is restricted from being used  no-restricted-properties

✖ 2 problems (2 errors, 0 warnings)
```

**What actually happened? Please include the actual, raw output from ESLint.**
```sh
<text>
  1:34  error  'foo.bar' is restricted from being used  no-restricted-properties

✖ 1 problem (1 error, 0 warnings)
```

Per https://github.com/eslint/eslint/pull/7139#issuecomment-246749114

**Comment by not-an-aardvark** [comment]
I have a fix for this, but I ended up building it on top of #7139 to avoid having to rewrite it when/if #7139 gets merged. Should I add the fix to #7139, or wait until #7139 is merged and then submit a PR afterwards?

**Comment by platinumazure** [comment]
My suggestion is write on top of the other PR and rebase when the other one is merged, but do what's easiest for you!

**Comment by not-an-aardvark** [comment]
Thanks, but I meant to ask whether I should add the fix to the #7139 branch, i.e. have the fix and the rest of #7139 be part of the same PR. I supposed it's probably better to wait for #7139 to get merged and then make a PR for the fix separately.

**Comment by platinumazure** [comment]
As long as you rebase after the original PR is merged, I think you can do
basically whatever you like.
